xbmc/FileItemList.cpp

```diff
@@ -705,7 +705,31 @@ void CFileItemList::RemoveExtensions()
     m_items[i]->RemoveExtension();
 }
 
-void CFileItemList::Stack(bool stackFiles /* = true */)
+void CFileItemList::ConvertDiscFoldersToFiles()
+{
+  for (const auto& item : m_items)
+  {
+    const std::string playPath = VIDEO::UTILS::GetOpticalMediaPath(*item);
+    if (!playPath.empty())
+      ConvertDiscFolderToFile(item, playPath);
+  }
+}
+
+void CFileItemList::ConvertDiscFolderToFile(const std::shared_ptr<CFileItem>& item,
+                                            const std::string& playPath)
+{
+
+  item->SetPath(playPath); // updated path (for DVD/Bluray files)
```

Comment:
Although we are doing it at several places for historical reasons, re-setting item's path after creation is a bug! The path is the identity of an item that must not change during entire lifetime of the instance. We introduced dynpath to keep what actually shall be played. And looking at the name of the variable `playPath` this reads like a perfect match for a dynpath.